ngraph_bridge/ngraph_builder.cc

```diff
@@ -1977,13 +1977,23 @@ static Status TranslateFillOp(
 static Status TranslateFloorDivOp(
     const Node* op, const std::vector<const Tensor*>& static_input_map,
     Builder::OpMap& ng_op_map) {
-  auto ng_floordiv = [&op](std::shared_ptr<ng::Node> ng_input1,
-                           std::shared_ptr<ng::Node> ng_input2) {
-    return ConstructNgNode<ng::op::Floor>(
-        op->name(),
-        ConstructNgNode<ng::op::Divide>(op->name(), ng_input1, ng_input2));
-  };
-  return TranslateBinaryOp(op, static_input_map, ng_op_map, ng_floordiv);
+  DataType dtype;
+  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "T", &dtype));
+  if (dtype == DT_INT32) {
+    auto ng_div = [&op](std::shared_ptr<ng::Node> ng_input1,
+                        std::shared_ptr<ng::Node> ng_input2) {
+      return ConstructNgNode<ng::op::Divide>(op->name(), ng_input1, ng_input2);
+    };
+    return TranslateBinaryOp(op, static_input_map, ng_op_map, ng_div);
+  } else {
+    auto ng_floordiv = [&op](std::shared_ptr<ng::Node> ng_input1,
+                             std::shared_ptr<ng::Node> ng_input2) {
+      return ConstructNgNode<ng::op::Floor>(
+          op->name(),
+          ConstructNgNode<ng::op::Divide>(op->name(), ng_input1, ng_input2));
+    };
+    return TranslateBinaryOp(op, static_input_map, ng_op_map, ng_floordiv);
+  }
```

Comment:
I wonder if a static analyzer like klocwork or some compiler might complain "not returning anything from a non-void function"

we can do a `return errors::Internal("We should never reach this part in translating floordiv");`

or we can declare function outside the ifelse `Status std::function<std::shared_ptr<ng::Node>, std::shared_ptr<ng::Node>>`, populate it inside the ifelse and then have a single line outside the if-else that says `return TranslateBinaryOp(op, static_input_map, ng_op_map, ng_bin_fn);`